apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/ErrorReporting/ErrorReportingExceptionLogger.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using Google.Cloud.ErrorReporting.V1Beta1;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+
+namespace Google.Cloud.Diagnostics.AspNetCore
+{
+    /// <summary>
+    ///  Google Cloud Error Reporting Exception Logger.
+    /// </summary>
+    /// 
+    /// <remarks>
+    /// Reports unhandled exceptions to Google Cloud Error Reporting.
+    /// Docs: https://cloud.google.com/error-reporting/docs/
+    /// </remarks>
+    public sealed class ErrorReportingExceptionLogger
+    {
+        /// <summary> The Google Cloud Platform project id.</summary>
+        private readonly ProjectName _projectName;
+
+        ///<summary>
+        /// The service context in which this error has occurred.
+        /// See: https://cloud.google.com/error-reporting/reference/rest/v1beta1/projects.events#ServiceContext
+        /// </summary>
+        private readonly ServiceContext _serviceContext;
+
+        /// <summary>The client to report errors with.</summary>
+        private readonly Task<ReportErrorsServiceClient> _clientTask;
+
+        /// <summary>
+        /// Creates an instance of <see cref="ErrorReportingExceptionLogger"/>
+        /// </summary>
+        /// <param name="client">The Error Reporting client.</param>
+        /// <param name="projectId">The Google Cloud Platform project ID.</param>
+        /// <param name="serviceName">An identifier of the service, such as the name of the executable or job.</param>
+        /// <param name="version">Represents the source code version that the developer provided.</param> 
```

Comment:
These docs should indicate which parameters must not be null and which can be null (same below)